src/player_activity.cpp

```diff
@@ -318,6 +318,11 @@ void player_activity::finish( player *p )
             p->do_read(&(p->i_at(position)));
             if (type == ACT_NULL) {
                 add_msg(_("You finish reading."));
+
+                // below code added in to support reading from ground/vehicles
+                if ( values[1] != FRM_INV ) {
+                    g->drop(p->inv.size()-1);
```

Comment:
I might have been unclear.  I'm not worried about p->do_read
assuming position is where the book is, I'm worried about g->drop assuming
inv.size()-1 is where the book is.

On Wednesday, September 30, 2015, Coolthulhu <notifications@github.com>
wrote:

> In src/player_activity.cpp
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/13674#discussion_r40794538>
> :
>
> > @@ -318,6 +318,11 @@ void player_activity::finish( player *p )
> >              p->do_read(&(p->i_at(position)));
> >              if (type == ACT_NULL) {
> >                  add_msg(_("You finish reading."));
> > +
> > +                // below code added in to support reading from ground/vehicles
> > +                if ( values[1] != FRM_INV ) {
> > +                    g->drop(p->inv.size()-1);
>
> do_read will most likely crash if the selected item is not a book.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/13674/files#r40794538>.
>